File: flang/test/OmpSs-RT/07_phases_ompss.dg/fortran/success_task_01.f90
! RUN: %oss-compile-and-run
! <testinfo>
! test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
! </testinfo>
PROGRAM P
    INTEGER :: X(20, 10)

    X = 1

    !$OSS TASK INOUT(X(2, 1))
       X(2, 1) = X(2, 1) + 41
    !$OSS END TASK


    !$OSS TASK IN(X(2, 1))
        IF (X(2,1) /= 42) STOP 1
    !$OSS END TASK

    !$OSS TASKWAIT
END PROGRAM P

File: clang/test/OmpSs-RT/07_phases_ompss_2.dg/c/success_task_01.c
// RUN: %oss-compile-and-run
// RUN: %oss-O2-compile-and-run
// UNSUPPORTED: true
/*
<testinfo>
test_generator="config/mercurium-ompss-2"
test_nolink=yes
</testinfo>
*/

#pragma oss task
void f()
{ }

void g() {
    #pragma oss task
    {
        f();
    }

    #pragma oss task for chunksize(4)
    for (int i = 0; i < 100; ++i)
    {}

    #pragma oss taskwait
}
